.github/workflows/pr-title.yaml

```diff
@@ -0,0 +1,23 @@
+name: lint-pr-title
+
+on:
+  pull_request:
+    branches:
+      - v2
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
+
+jobs:
+  conventional-commit:
+    if: github.repository == 'SeldonIO/seldon-core' # Do not run this on forks.
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    steps:
+      - name: Check PR title matches Conventional Commit format
+        uses: matthiashermsen/lint-pull-request-title@v1.0.0
+        with:
+          allowed-pull-request-types: build,ci,docs,feat,fix,refactor,revert,test
```

Comment:
if we have a developer/contribution guide section, perhaps add this to it?

Reply:
We do indeed have a `CONTRIBUTING.md`.  I'll enact this shortly - good suggestion!

Reply:
Fixed in https://github.com/SeldonIO/seldon-core/pull/4891/commits/adc91d0502b89383a8d5f42ab2f7122153da0632